### PR TITLE
Core: migrate start_inventory_from_pool to new options API

### PR DESCRIPTION
**Pull request by alwaysintreble**
## What is this fixing or adding?
I'm really good at missing things

### CODE REVIEW

**Review by ScipioWright — APPROVED** [review]
Tested by adding items to start_inventory_from_pool in Tunic. It functioned as normal, except without giving warnings about getting options from multiworld being deprecated.

File: Main.py [hunk 1]
```diff
@@ -167,10 +169,12 @@ def main(args, seed=None, baked_server_options: Optional[Dict[str, object]] = No
 
     # remove starting inventory from pool items.
     # Because some worlds don't actually create items during create_items this has to be as late as possible.
-    if any(world.start_inventory_from_pool[player].value for player in world.player_ids):
+    if any(hasattr(world.worlds[player].options, "start_inventory_from_pool") for player in world.player_ids):
```
Comment (black-sliver) [inline Main.py:172]: I think this change is not fully correct. the old code skips for `None` as well as `{}` , the new code will run for `{}`. Should probably be `any(getattr(..., ..., None) for ...)`

File: Main.py [hunk 2]
```diff
@@ -167,10 +169,12 @@ def main(args, seed=None, baked_server_options: Optional[Dict[str, object]] = No
 
     # remove starting inventory from pool items.
     # Because some worlds don't actually create items during create_items this has to be as late as possible.
-    if any(world.start_inventory_from_pool[player].value for player in world.player_ids):
+    if any(getattr(world.worlds[player].options, "start_inventory_from_pool", None) for player in world.player_ids):
         new_items: List[Item] = []
         depletion_pool: Dict[int, Dict[str, int]] = {
-            player: world.start_inventory_from_pool[player].value.copy() for player in world.player_ids}
+            player: getattr(world.worlds[player], "start_inventory_from_pool", StartInventoryPool({})).value.copy()
```
Comment (black-sliver) [inline Main.py:175]: this is missing the `.options`, so it'll always return empty. I think going forward we should maybe add optional core options to the dataclass somehow, but that's out of scope here.
```suggestion
            player: getattr(world.worlds[player].options,
                            "start_inventory_from_pool",
                            StartInventoryPool({})).value.copy()
```

**Review by black-sliver — CHANGES_REQUESTED** [review]

**Review by black-sliver — APPROVED** [review]
Thanks!

**Pull request merged**